README.md

```diff
@@ -18,3 +18,45 @@ Resume a training run specified by `<config_file>` starting from `<run_id>/<chec
 ```
 python src/ --config conf/train/<config_file>.yml --resume models/<run_id>/<checkpoint_file>
 ```
+
+
+## Extending this repo
+
+We follow the following flow for objects in this repository:
+![image of the flow map](<readme_assets/flow map.png>)
+
+These initialized classes are passed the dictionary parsed directly from yaml as arguments. We recommend reading examples for your respective function class in [conf/](https://github.com/in-context-learning-2024/in-context/blob/main/conf/)
+
+Make sure to initialize your parent class with `super().__init__(...)` in both cases
+
+We try to make it easy to contribute by [adding models](#adding-models) or [function classes](#adding-function-classes):
+
+
+---
+### Adding Models
+
+We wrap all sequence architectures and baselines as [`ContextModel`s](https://github.com/in-context-learning-2024/in-context/blob/main/src/core/context_model.py). The order of calls against a `ContextModel` is as follows:
+- We `__init__` the `ContextModel` directly from the yaml configuration
+- Inside the training loop and during evaluation, we call `forward` to generate predictions
+
+`ContextModel`s thus have only two required methods:
+- `__init__(...)`, and
+- `forward(self, xs: torch.Tensor, ys: torch.Tensor) -> torch.Tensor`
+
+In addition, we require both `.context_length: int` and `.name: str` to be set for evaluation and visualization.
+
+---
+### Adding Function Classes
+
+We take the approach of framing each function class as an iterator, where the boilerplate of making the iterator work for our training scheme (`ContextTrainer`) is [implemented for you](https://github.com/in-context-learning-2024/in-context/blob/main/src/core/function_class.py). The order of calls against a `FunctionClass` is as follows:
+- We `__init__` the `FunctionClass` with the parsed yaml configuration
+    - this calls `_init_param_dist()`, which should produce a [`torch.distributions.Distribution`](https://pytorch.org/docs/stable/distributions.html#distribution) 
+        - Note: If the implemented ones in pytorch are not enough, we also provide a way to ["combine" distributions](https://github.com/in-context-learning-2024/in-context/blob/main/src/utils.py)
```

Comment:
change to "Note: If the distributions implemented in PyTorch are not enough, ..."

Reply:
done.